homeassistant/components/sensor/tellduslive.py

```diff
@@ -46,70 +46,97 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     """ Sets up Tellstick sensors. """
     if discovery_info is None:
         return
-    sensors = tellduslive.NETWORK.get_sensors()
-    devices = []
-
-    for component in sensors:
-        for sensor in component["data"]:
-            # one component can have more than one sensor
-            # (e.g. both humidity and temperature)
-            devices.append(TelldusLiveSensor(component["id"],
-                                             component["name"],
-                                             sensor["name"]))
-    add_devices(devices)
+    add_devices(TelldusLiveSensor(sensor) for sensor in discovery_info)
 
 
 class TelldusLiveSensor(Entity):
     """ Represents a Telldus Live sensor. """
 
-    def __init__(self, sensor_id, sensor_name, sensor_type):
-        self._sensor_id = sensor_id
-        self._sensor_type = sensor_type
-        self._state = None
-        self._name = "{} {}".format(sensor_name or DEVICE_DEFAULT_NAME,
-                                    SENSOR_TYPES[sensor_type][0])
-        self._last_update = None
-        self._battery_level = None
+    def __init__(self, sensor_id):
+        self._id = sensor_id
         self.update()
+        _LOGGER.debug("created sensor %s", self)
+
+    def update(self):
+        """ update sensor values """
+        _LOGGER.debug("updating sensor %s", self)
+        tellduslive.NETWORK.update_sensors()
+
+    @property
+    def _sensor(self):
+        """ return the local representation of the sensor state """
+        return tellduslive.NETWORK.get_sensor(self._id)
+
+    @property
+    def _sensor_name(self):
+        return self._sensor["name"]
+
+    @property
+    def _sensor_value(self):
+        return self._sensor["data"]["value"]
+
+    @property
+    def _sensor_type(self):
+        return self._sensor["data"]["name"]
+
+    @property
+    def _sensor_battery_level(self):
+        return self._sensor.get("battery")
+
+    @property
+    def _sensor_last_updated(self):
+        return self._sensor.get("lastUpdated")
+
+    @property
+    def _battery_level(self):
```

Comment:
Is ie really necessary to have properties to free refernce other properties? If all these things are used once we might as well inline them